app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -0,0 +1,50 @@
+module ForemanDeployments
+  module Tasks
+    class SearchTaskDefinition < BaseDefinition
+      class Action < BaseAction
+        def run
+          results = SearchTaskDefinition.search(input)
+          results = SerializableArray.new(results)
+          SearchTaskDefinition.create_output(results, output)
+        end
+      end
+
+      class SerializableArray < Array
+        def to_hash
+          {}
+        end
+      end
+
+      def validate
+        results = SearchTaskDefinition.search(parameters)
+        messages = nil
+        unless results.any?
+          messages = ["#{parameters[:klass]}.search_for(\"#{parameters[:search_term]}\") didnt return valid objects"]
```

Comment:
My previous comment got lost with your update: Validation messages are presented to users. Therefore I think they should be translated.